packages/core/content-manager/admin/src/preview/components/PreviewHeader.tsx

```diff
@@ -0,0 +1,208 @@
+
+import { getDocumentStatus } from '../../pages/EditView/EditViewPage';
+import { usePreviewContext } from '../pages/Preview';
+
+/* -------------------------------------------------------------------------------------------------
+ * ClosePreviewButton
+ * -----------------------------------------------------------------------------------------------*/
+
+const ClosePreviewButton = () => {
+  const [{ query }] = useQueryParams();
+  const navigate = useNavigate();
+  const { formatMessage } = useIntl();
+
+  const canGoBack = useHistory('BackButton', (state) => state.canGoBack);
+  const goBack = useHistory('BackButton', (state) => state.goBack);
+  const history = useHistory('BackButton', (state) => state.history);
+
+  const fallbackUrl: To = {
+    pathname: '..',
+    search: stringify(query, { encode: false }),
+  };
+
+  const handleClick = (e: React.MouseEvent) => {
+    /**
+     * Prevent normal link behavior. We only make it an achor for accessibility reasons.
+     * The point of this logic is to act as the browser's back button when possible, and to fallback
+     * to a link behavior to the edit view when no history is available.
+     *  */
+    e.preventDefault();
+
+    if (canGoBack) {
+      goBack();
+    } else {
+      navigate(fallbackUrl);
+    }
+  };
+
+  return (
+    <IconButton
+      tag={Link}
+      to={history.at(-1) ?? fallbackUrl}
+      onClick={handleClick}
+      label={formatMessage({
+        id: 'content-manager.preview.header.close',
+        defaultMessage: 'Close preview',
+      })}
+    >
+      <Cross />
+    </IconButton>
+  );
+};
+
+/* -------------------------------------------------------------------------------------------------
+ * DocumentStatus
+ * -----------------------------------------------------------------------------------------------*/
+
+interface StatusData {
+  background: BoxProps['background'];
+  border: BoxProps['borderColor'];
+  text: BoxProps['color'];
+  message: MessageDescriptor;
+}
+
+const getStatusData = (status: ReturnType<typeof getDocumentStatus>): StatusData => {
+  switch (status) {
+    case 'draft':
+      return {
+        background: 'secondary100',
+        border: 'secondary200',
+        text: 'secondary700',
+        message: {
+          id: 'content-manager.containers.List.draft',
+          defaultMessage: 'Draft',
+        },
+      };
+    case 'modified':
+      return {
+        background: 'alternative100',
+        border: 'alternative200',
+        text: 'alternative700',
+        message: {
+          id: 'content-manager.containers.List.modified',
+          defaultMessage: 'Modified',
+        },
+      };
+    case 'published':
+    default:
+      return {
+        background: 'success100',
+        border: 'success200',
+        text: 'success700',
+        message: {
+          id: 'content-manager.containers.List.published',
+          defaultMessage: 'Published',
+        },
+      };
+  }
+};
+
```

Comment:
How come we have to do this here specifically for preview? Isn't this similar to what we are doing in history and maybe elsewhere in the app?

https://github.com/strapi/strapi/blob/11331d7572f8d2da397e6268b4f2d82f836e0998/packages/core/content-manager/admin/src/history/components/VersionsList.tsx#L45

Is there a difference? Or would it make sense to share this function instead of duplicating it?

Reply:
Yes that's the code I copied. In the Content Manager list and edit view it uses [a DocumentStatus component](https://github.com/strapi/strapi/blob/2f7b9ae55adbbf4416af874fb13b222b851c4432/packages/core/content-manager/admin/src/pages/EditView/components/DocumentStatus.tsx#L19). It relies on [Status from the design system](https://design-system.strapi.io/?path=/docs/design-system-components-status--docs) to apply the right text/border/bg colors.

I'd like reuse Document status across the app, but Status from the DS doesn't have a variant small enough to match our designs. So we need to open a PR in the DS, ship it, bump it in Strapi, update DocumentStatus, then we can delete the extraneous code both here and in history.

In the meantime we're left with this 🤷

Reply:
I'm working on opening the DS PR now so we don't forget